cmd/kubeadm/app/cmd/join.go

```diff
@@ -179,23 +183,51 @@ func NewCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
 	if joinOptions == nil {
 		joinOptions = newJoinOptions()
 	}
+	joinRunner := workflow.NewRunner()
 
 	cmd := &cobra.Command{
 		Use:   "join",
 		Short: "Run this on any machine you wish to join an existing cluster",
 		Long:  joinLongDescription,
 		Run: func(cmd *cobra.Command, args []string) {
-			joinData, err := newJoinData(cmd, args, joinOptions, out)
+			joinOptions.args = &args
```

Comment:
How does behave the runner together with args? I'm mostly concerned about [this]( https://github.com/kubernetes/kubernetes/blob/4531ce692a3557f2c1985409480095d7c884c341/cmd/kubeadm/app/cmd/phases/workflow/runner.go#L308)

Reply:
did some basic tests and couldn't see a problem with the `cobra.NoArgs` of the sub command.
we should test again once there are actual phases...